File: app/src/main/java/com/example/todonotesapp/adapter/NotesAdapter.java
package com.example.todonotesapp.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.example.todonotesapp.R;
import com.example.todonotesapp.clicklistener.ItemClickListener;
import com.example.todonotesapp.model.Notes;
import java.util.List;

public class NotesAdapter extends RecyclerView.Adapter<NotesAdapter.ViewHolder> {

    private List<Notes> listNotes;
    private ItemClickListener itemClickListener;

    // to get list in adapter file create a constructor of this NoteAdapter
    public NotesAdapter(List<Notes> list, ItemClickListener itemClickListener){

        this.listNotes = list;
        this.itemClickListener = itemClickListener;
    }

    @NonNull
    @Override
    public NotesAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        // set a card layout on viewHolder si it's make a card
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.notes_adapter_layout,parent,false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull NotesAdapter.ViewHolder holder, int position) {
        // we bind the data from list to viewHolder
        final Notes notes = listNotes.get(position);

        // get data from list
        String title = notes.getTitle();
        String description = notes.getDescription();

        // set date on card
        holder.textViewTitle.setText(title);
        holder.textViewDescription.setText(description);

        // assign set onClickListener on item
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                itemClickListener.onClick(notes);
            }
        });


    }

    @Override
    public int getItemCount() {
        return listNotes.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder{
        TextView textViewTitle, textViewDescription;

        // this is a constructor of viewHolder its work on
        // findViewById on(create in card layout fields ) variables
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            textViewTitle = itemView.findViewById(R.id.textViewTitle);
            textViewDescription = itemView.findViewById(R.id.textViewDescription);
        }
    }
}

File: app/src/main/java/com/example/todonotesapp/adapter/NotesAdapter.kt
package com.example.todonotesapp.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.todonotesapp.R
import com.example.todonotesapp.clicklistener.ItemClickListener
import com.example.todonotesapp.model.Notes

class NotesAdapter (val list: List<Notes>,val itemClickListener:ItemClickListener)
    :RecyclerView.Adapter<NotesAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NotesAdapter.ViewHolder {

        val view = LayoutInflater.from(parent.context).inflate(R.layout.notes_adapter_layout,parent,false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: NotesAdapter.ViewHolder, position: Int) {

        // get list and its variable's position
        val notes = list[position]

        //variables get data from list using position
        val title = notes.title
        val description = notes.description

        // setting up data to recyclerView card using upper variables
        holder.textViewTitle.text = title
        holder.textViewDescription.text = description

        // assign setOnClickListener on card;s item
        holder.itemView.setOnClickListener(object : View.OnClickListener{
            override fun onClick(v: View?) {
                itemClickListener.onClick(notes)
            }

        })
    }

    override fun getItemCount(): Int {

        return list.size
    }

    class ViewHolder(itemView:View) :RecyclerView.ViewHolder(itemView) {

        // this is a constructor of viewHolder its work on
        // findViewById on(create in card layout fields ) variables
        val textViewTitle :TextView = itemView.findViewById(R.id.textViewTitle)
        val textViewDescription :TextView = itemView.findViewById(R.id.textViewDescription)
    }
}